src/p4pktgen/config.py

```diff
@@ -116,3 +117,8 @@ def get_do_consolidate_tables(self):
 
     def get_randomize(self):
         return self.randomize
+
+    def get_extern_definitions(self):
+        if self.extern_definitions is None:
+            return []
```

Comment:
Should this be `{}` rather than `[]`?

Reply:
Well spotted.  Doesn't cause any problems as it turns out, but yes, it should be an empty dict.